dask_ml/model_selection/_incremental.py

```diff
@@ -347,3 +356,302 @@ def fit(
+
+        cv_results = {
+            "params": [x["params"] for v in info for x in v],
+            "test_score": scores,
+            "mean_test_score": scores,  # for sklearn comptability
+            "rank_test_score": ranks,
+            "mean_partial_fit_time": np.array(
+                [np.mean(x["partial_fit_time"]) for v in info for x in v]
+            ),
+            "std_partial_fit_time": np.array(
+                [np.std(x["partial_fit_time"]) for v in info for x in v]
+            ),
+            "mean_score_time": np.array(
+                [np.mean(x["score_time"]) for v in info for x in v]
+            ),
+            "std_score_time": np.array(
+                [np.std(x["score_time"]) for v in info for x in v]
+            ),
+            "partial_fit_calls": [x["partial_fit_calls"] for v in info for x in v],
+            "model_id": model_ids,
+        }
+        return cv_results
+
+    def _get_best(self, results, cv_results):
+        # type: (Dict, Dict) -> Estimator
+        """Select the best estimator from the set of estimators."""
+        rank = cv_results["rank_test_score"]
+        best_index = np.arange(len(rank))[rank - 1][0]
+        best_model_id = cv_results["model_id"][best_index]
+        assert cv_results["test_score"][best_index] == max(cv_results["test_score"])
+
+        loop_id, model_id = map(int, best_model_id.split("-"))
+        return results[loop_id][1][model_id].result(), best_index
+
+    def _update_results(self, results):
+        # type: (Dict) -> None
+        """Update `self` with attributes extracted from `results`."""
+        pass
+
+    def _process_results(self, results):
+        """Called with the output of `fit` immediately after it finishes.
+
+        Subclasses may update the results here, before further results are
+        computed (e.g. ``cv_results_``, ``best_estimator_``).
+
+        By default, results is returned as-is.
+        """
+        return results
+
+    def fit(self, X, y, **fit_params):
+        """Find the best parameters for a particular model
+
+        Parameters
+        ----------
+        X, y : array-like
+        **fit_params
+            Additional partial fit keyword arguments for the estimator.
+        """
+        ids_and_args = self._get_ids_and_args()
+        X, y = check_array(X, y)
+
+        X_train, X_test, y_train, y_test = self._get_cv(X, y)
+        scorer = check_scoring(self.estimator, scoring=self.scoring)
+
+        results = {
+            model_id: fit(
+                self.estimator,
+                param_list,
+                X_train,
+                y_train,
+                X_test,
+                y_test,
+                additional_calls=getattr(additional_calls, "fit", additional_calls),
+                fit_params=fit_params,
+                scorer=scorer,
+                random_state=self.random_state,
+            )
+            for model_id, (param_list, additional_calls) in ids_and_args.items()
+        }
```

Comment:
The Hyperband implementation is weird in that they'll want to call fit several times rather than just once (which I think will be more common).  Additionally, Hyperband will want to call fit many times concurrently, so we'll need to use coroutines for this.  

Short term I suggest that we implement classes for fitting one model,  RandomSearch, and SuccessiveHalving using the functions found in [this notebook](https://gist.github.com/mrocklin/4c95bd26d15281d82e0bf2d27632e294).  Then we look at Hyperband as a special case that has to call fit a few times and see what extra methods might have to be pulled out.

I'm happy to take a look at this later today as well if that would be useful.

Reply:
In short, I think that this approach is more complicated than necessary specifically because it is targetting Hyperband first and because Hyperband is a bit of a special case.  I think that we should target some of the simpler situations first, build conventions around them, and then see how we can add release valves for Hyperband to also work.

Reply:
My reply seems to have been lost.

tldr: that seems sensible. I agree that Hyperband's multiple calls to fit complicated this PR, and I don't know how representative that will be of other strategies (I assume not).